### PR TITLE
Support fauxton on IE10,IE11 and Edge

### PR DESCRIPTION
**Pull request by popojargo**
## Overview

- Added babel-polyfill to support ES6 functions on browsers that doesn't support them
- Fixed layouts problems on IE11 and Edge

## Testing recommendations

Since we seems to mainly support Firefox and Chrome, there are not automated tests for IE.

With my changes, I fixed:
- EDGE: Fixed JSON API bar display
- IE11/IE10 : Fixed replication page that wasnt shown
- IE11/IE10 : We are now able to use React 👍 
- IE11/IE10 : Polling slider where not shown correctly (but they still don't work correctly on IE)

## GitHub issue number

#949

## Checklist

- [x] Code is written and works correctly;
- [x] Changes are covered by tests;
- [ ] Documentation reflects the changes;
- [ ] Update [rebar.config.script](https://github.com/apache/couchdb/blob/master/rebar.config.script) with the correct tag once a new Fauxton release is made

### CODE REVIEW

**Comment by Antonio-Maranhao** [comment]
@popojargo @garrensmith 
Took me some time to get a VM with IE11 but now I'm set. I'll continue looking into this tomorrow.

File: webpack.config.dev.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ const settings = require('./tasks/helper')
 
 module.exports = {
   entry: {
-    bundle: './app/main.js' //Our starting point for our development.
+    bundle: ['babel-polyfill','./app/main.js'] //Our starting point for our development.
```
Comment (Antonio-Maranhao) [inline webpack.config.dev.js]: To address a previous comment from @garrensmith we could be more granular using `core-js` and only polyfill only what's needed. I was able to launch the UI using `bundle: ['core-js/fn/array', 'core-js/fn/symbol', 'core-js/fn/promise', './app/main.js']`

Comment (Antonio-Maranhao) [inline webpack.config.dev.js]: Side note... `webpack.config.release.js` needs to be updated as well.

File: app/addons/activetasks/components.js [hunk 2]
```diff
@@ -388,8 +388,13 @@ export class ActiveTasksPollingWidgetController extends React.Component {
   state = this.getStoreState();
 
   render() {
+    const isIE1X = document.documentMode == 11 || document.documentMode == 10;
```
Comment (Antonio-Maranhao) [inline app/addons/activetasks/components.js]: Could you replace with the `Helpers.isIE1X()` function you created?

**Comment by Antonio-Maranhao** [comment]
Btw... @popojargo are you planning on fixing Polling slider as well? If not we should open an issue so we don't lose track of it.

**Comment by garrensmith** [comment]
We using Bluebird for promises so we might not need that polyfill.

All misspelling thanks to my iPhone.
________________________________

File: webpack.config.dev.js (hunk 1 above)
Comment (popojargo) [reply webpack.config.dev.js]: Do you have any idea how I can found which polyfill are missings?

**Comment by popojargo** [comment]
@Antonio-Maranhao As for the polling, I think it could be for another issue. This is mostly a problem with the mouse events in IE and doesn't have any easy fix so far.

**Comment by popojargo** [comment]
@Antonio-Maranhao I managed to fix the issue for the polling with a package: react-script. The onChange event for range inputs should be fixed on React16(but I can't confirm).

**Comment by Antonio-Maranhao** [comment]
@garrensmith any concerns about the ISC license for `react-range`?
https://github.com/mapbox/react-range

**Comment by popojargo** [comment]
@Antonio-Maranhao ISC is like MIT so it should pass right?

**Comment by garrensmith** [comment]
From the list it looks fine https://www.apache.org/legal/resolved.html#category-x

All misspelling thanks to my iPhone.
________________________________

**Pull request merged**